eventing/samples/github-events/README.md

```diff
@@ -0,0 +1,162 @@
+`Flow` resource takes care of provisioning the webhook, the `Service`, the 
+`Channel`, and the `Subscription`.
+
+In response to a pull request event, the sample app _legit_ Service will add 
+`(looks pretty legit)` to the PR title.
+
+## Prerequisites
+
+- A Kubernetes cluster with Knative serving installed. Follow the
+  [installation instructions](https://github.com/knative/docs/blob/master/install/README.md)
+  if you need to create one.
+- [Docker](https://www.docker.com/) installed and running on your local machine,
+  and a Docker Hub account configured (you'll use it for a container registry).
+- The Kubernetes cluster also has Knative eventing core installed. You can
+  install with:
+  ```shell
+  kubectl apply -f https://storage.googleapis.com/knative-releases/eventing/latest/release.yaml
+  ```
+- A domain name that allows GitHub to call into the cluster: Follow the
+  [configure a custom domain](https://github.com/knative/docs/blob/master/serving/using-a-custom-domain.md) and 
+  [assign a static IP address](https://github.com/knative/docs/blob/master/serving/gke-assigning-static-ip-address.md)
+  instructions.
+
+## Configuring Knative
+
+To use this sample, you'll need to install the `stub` ClusterBus and the 
+`github` EventSource.
+
+```shell
+kubectl apply -f https://storage.googleapis.com/knative-releases/eventing/latest/release-clusterbus-stub.yaml
+kubectl apply -f https://storage.googleapis.com/knative-releases/eventing/latest/release-source-github.yaml
+```
+
+## Granting permissions
+
+Because the `github` EventSource needs to create a Knative Service, you'll need
+to provision a special ServiceAccount with the necessary permissions.
+
+The `eventing/samples/github-events/auth.yaml` file provisions a service
+account, and creates a role which can create a Knative Service in the `default`
+namespace. In a production environment, you might want to limit the access of
+this service account to only specific namespaces.
+
+```shell
+kubectl apply -f eventing/samples/github-events/auth.yaml
+```
+
+## Build and deploy the sample
+
+1.  Use Docker to build the sample code into a container. To build and push with
+    Docker Hub, run these commands replacing `{username}` with your Docker Hub
+    username. Run the following from the _root_ of the `knative/docs` repo:
+
+    ```shell
+    # Build the container on your local machine
+    docker build -t {username}/github-events --file=eventing/samples/github-events/Dockerfile .
+
+    # Push the container to docker registry
+    docker push {username}/github-events
+    ```
+
+1.  After the build has completed and the container is pushed to Docker Hub, you
+    can deploy the function into your cluster. **Ensure that the container image
+    value in `function.yaml` matches the container you built in the previous
+    step.** Apply the configuration using `kubectl`:
+
+    ```shell
+    kubectl apply -f eventing/samples/github-events/function.yaml
```

Comment:
Again (and below), it would be nice if these could be relative to the current directory.

I'll see if I can do this for the `k8sevents` sample once I get my laptop back this afternoon.

Reply:
Added an issue to fix this for all eventing samples: https://github.com/knative/docs/issues/351